addons/sale_project/tests/test_sale_project.py

```diff
@@ -652,3 +652,98 @@ def test_quick_create_sol(self):
         sale_line = self.env['sale.order.line'].browse(sale_line_id)
         self.assertEqual(sale_line.product_id, product_service, 'The created SOL should use the right product.')
         self.assertTrue(product_service.name in sale_line_name, 'The created SOL should use the full name of the product and not just what was typed.')
+
+    def test_sale_order_with_project_task_from_multi_companies(self):
+        SaleOrder = self.env["sale.order"]
+        Company = self.env["res.company"]
+        Product = self.env["product.product"]
+        Project = self.env["project.project"]
+        Partner = self.env["res.partner"]
+
+        uom_hour = self.env.ref("uom.product_uom_hour")
+        will_smith = Partner.create({"name": "Will Smith"})
+        multi_company_project = Project.create({
```

Comment:
`Project` and `Partner` are just used once so we could directly use `self.env[...]` instead. 🙂 

```suggestion
        uom_hour = self.env.ref("uom.product_uom_hour")
        will_smith = self.env["res.partner"].create({"name": "Will Smith"})
        multi_company_project = self.env["project.project"].create({
```